src/test/skript/tests/syntaxes/effects/EffParse.sk

```
@@ -0,0 +1,32 @@
+test "parse effect":
+	# test single value
+	set {_value} to "5"
+	parse {_value} as int
+	assert {_value} = 5 with "couldn't parse a single value"
+
+	set {_value} to "a"
+	parse {_value} as int
+	assert {_value} isn't set with "failed parse didn't delete the value"
```

Should variables that fail to parse be deleted? I can see this being a bit annoying.

---

it mimics the behavior of the parse expression

---

I guess, but in the expression the user would have to do `set x to x parsed as y` to get that, right? And there's a way around it by just setting another var to the parsed value and checking if it's set, which you cant do with the effect. The explanation isnt the greatest but I think it should silently fail (or loudly warn?) if it doesnt work.

---

what about not deleting it but having `last parse error` set

---

I think setting it to null is best
if you're parsing "test" as an int, having it still be a string is not a great result.
Consider `"1 "`, debugging that before and after parsing looks the same as if it were `"1"`, IF you don't set it to null
If you set the result to null, then it's clear the parsing failed.

---

Yeah that works, though I think I'd prefer a runtime warning or something. Though I can see that getting a bit messy.

---

That said, perhaps an option like `try to parse x as y` could be useful to have the non-null behavior?

---

Big fan of this yeah. `try to` seems like the best option to me actually.